### PR TITLE
Add additional buffer types

### PR DESCRIPTION
**Pull request by ahmed-adam**
This change adds support for 3 additional buffer types which are present in proxy-wasm-rust-sdk

### CODE REVIEW

**Comment by PiotrSikora** [comment]
Could you rebase/merge on top of current `main` to re-kick the CI? Thanks!

File: Cargo.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-wasmtime = "11.0.1"
+wasmtime = "22.0.0"
```
Comment (PiotrSikora) [inline Cargo.toml:10]: This should be a separate PR, since it's unrelated to the buffer types change.

This is already done in #20, but needs updating...

Comment (ahmed-adam) [reply Cargo.toml:10]: OK, thanks @PiotrSikora . I will wait for #20 to be merged, then pull the changes into this PR.

Comment (PiotrSikora) [reply Cargo.toml:10]: It's merged now. Could you update this PR?

Comment (ahmed-adam) [reply Cargo.toml]: Done. Thanks

**Comment by PiotrSikora** [comment]
(the examples test is failing because of changes in Rust v1.80, fixed in #23).

**Review by PiotrSikora — APPROVED** [review]
Thanks!
